vignettes/admiral.Rmd

```diff
@@ -163,120 +97,157 @@ and `TRTSTMF` variables.
+    filter_add = DSDECOD == "FINAL LAB VISIT"
+  )
+```
 
-```{r}
-is.na(visits)
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(STUDYID, USUBJID, FINLABDT)
+)
 ```
 
-Thus, to filter all visits which are not `"Baseline"` the following condition would need to be used.
+## Filter Functions
 
-```{r}
-visits != "Baseline" | is.na(visits)
-```
+[Filter functions](../reference/index.html#utilities-for-filtering-observations) are `{admiral}`
+utilities that filter the input dataset in different manners, for instance returning records 
+that fit/don't fit a certain condition or that are the first/last observation in a by group.
+These functions form an important internal backbone to some of the `{admiral}` functions, but
+can also be used on their own to explore or manipulate a dataset. For instance, in the example 
+below we use `filter_extreme()` to extract the most recent MAP records in `advs`:
 
-Also note that most aggregation functions, like `mean()` or `max()`, also return `NA` if any element of the input vector is missing.
+```{r, eval = TRUE}
+advs_lastmap <- advs %>%
+  filter(PARAMCD == "MAP") %>%
+  filter_extreme(
+    by_vars = exprs(USUBJID),
+    order = exprs(AVISITN, PARAMCD),
+    mode = "last"
+  )
+```
 
-```{r}
-mean(c(1, NA, 2))
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  advs_lastmap,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVAL)
+)
 ```
 
-To avoid this behavior one has to explicitly set `na.rm = TRUE`.
+# Argument Conventions 
 
-```{r}
-mean(c(1, NA, 2), na.rm = TRUE)
-```
+Within the `{admiral}` package, any arguments which expect variable names or expressions of variable names, symbols or 
+expressions must be specified rather than strings.
 
-This is very important to keep in mind when using `{admiral}`'s aggregation functions such as `derive_summary_records()`.
+- For arguments which expect a single variable name, the name can be specified without quotes and quotation, e.g.
+`new_var = TEMPBL`
 
-For handling of `NA`s in sorting variables see [Sort
-Order](generic.html#sort_order).
+- For arguments which expect one or more variable names, a list of symbols is expected, e.g. `by_vars = exprs(PARAMCD, AVISIT)` 
 
-# Validation
+- For arguments which expect a single expression, the expression needs to be passed "as is", e.g. `filter = PARAMCD == "TEMP"`
 
-All functions are reviewed and tested to ensure that they work as described in
-the documentation. They are **not validated** yet.
+- For arguments which expect one or more expressions, a list of expressions is expected, e.g. `order = exprs(AVISIT, desc(AESEV))`
 
-Although `{admiral}` follows CDISC standards, it does not claim that the dataset
-resulting from calling `{admiral}` functions is ADaM compliant. This has to be
-ensured by the user.
+If you are new to expressions, consider reading our [Introduction to expressions](concepts_conventions.html#exprs) to
+learn more.
```

Comment:
Should we just use the name of the vignette section?

Reply:
changed